AEPAssurance/Source/QuickConnect/QuickConnectView.swift

```diff
@@ -0,0 +1,397 @@
+            self.connectButton.isUserInteractionEnabled = true
+        }
+    }
+    
+    ///
+    /// Sets the QuickConnectView to a waiting state where the connect button text changes to "Waiting..." and user interaction is disabled on it. It also hides any error dialogues that were previously present
+    ///
+    func waitingState() {
+        DispatchQueue.main.async {
+            self.errorTitle.isHidden = true
+            self.errorDescription.isHidden = true
+            self.connectButton.setTitle("Waiting...", for: .normal)
+            self.connectButton.backgroundColor = UIColor(red: 67.0/256.0, green: 67.0/256.0, blue: 67.0/256.0, alpha: 1)
+            self.connectButton.isUserInteractionEnabled = false
+        }
+    }
+    
+    ///
+    /// Sets the QuickConnectView to a successful connection state, where the connect button changes to "Connected" and user interaction on the button is disabled
+    ///
+    func connectionSuccessfulState(){
+        DispatchQueue.main.async {
+            self.connectButton.setTitle("Connected", for: .normal)
+            self.connectButton.backgroundColor = UIColor(red: 45.0/256.0, green: 157.0/256.0, blue: 120.0/256.0, alpha: 1)
+            self.connectButton.isUserInteractionEnabled = false
+        }
+    }
+    
+    ///
+    /// Sets the QuickConnectView to an error state, where the error dialogue is displayed with a given error, and the connect button
+    /// changes to "Retry"
+    ///
+    func errorState(errorText: String) {
+        DispatchQueue.main.async {
+            self.errorTitle.isHidden = false
+            self.errorDescription.isHidden = false
+            self.errorDescription.text = errorText
+            self.connectButton.setTitle("Retry", for: .normal)
+            self.connectButton.backgroundColor = UIColor(red: 20.0/256.0, green: 115.0/256.0, blue: 230.0/256.0, alpha: 1)
+            self.connectButton.isUserInteractionEnabled = true
+        }
+    }
+    
+    ///
+    /// Dismisses the QuickConnectView with an animation
+    ///
+    func dismiss() {
+        DispatchQueue.main.async {
+            guard let window = UIApplication.shared.assuranceGetKeyWindow() else {
+                return
+            }
+            
+            UIView.animate(withDuration: 0.3,
+                           delay: 0,
+                           options: [.curveEaseInOut],
+                           animations: { [self] in
+                self.baseView.frame.origin.y = window.frame.size.height
+            }, completion: { [self] _ in
+                self.baseView.removeFromSuperview()
+                self.displayed = false
+            })
+        }
+    }
+    
+    ///
+    /// Sets up the QuickConnectView subviews / constraints / and stackViews with the given UIWindow
+    /// - Parameter: window `UIWindow` the window to be used as the foundation
+    ///
+    func setupLayout(with window: UIWindow) {
+        window.addSubview(baseView)
```

Comment:
Does this account for safe area layouts or are we expecting the hardcoded margins from the AssuranceConstants to cover for this?

Reply:
We were using hardcoded values to account for this, but it is actually creating weird space on the bottom on devices which don't have safe area guides. I will add this. Good catch.

Reply:
![Simulator Screen Shot - iPhone 14 - 2023-03-09 at 10 47 16](https://user-images.githubusercontent.com/10753118/224112331-428ef024-97b6-489f-a067-a03a3d734221.png)
![Simulator Screen Shot - iPhone SE (3rd generation) - 2023-03-09 at 10 47 21](https://user-images.githubusercontent.com/10753118/224112339-6e847eba-89ea-4339-962e-ffb80af6460b.png)


Here are some screenshots, now using the safe area layout guide for the adobe logo.

Reply:
Looks good -- should we also do the same for left/right and top for completeness?

Reply:
@pfransenadb I don't think we need to be concerned about the top / left / right because this view doesn't have any navigation bars / tool bars on top.

Reply:
While it's a lower risk (given that this is debug mode, etc) -- I think it'd be preferable to obey safe area insets on all margins and not have to worry about this in the future.

Reply:
@pfransenadb Agree. I've gone ahead and added em.